community/kernel/src/main/java/org/neo4j/kernel/impl/store/CommonAbstractStore.java

```diff
@@ -282,6 +282,10 @@ private void extractHeaderRecord() throws IOException
                                     pageCache.pageSize() + " bytes.");
                         }
                     }
+                    else
+                    {
+                        throw new StoreNotFoundException( "Store file not found: " + storageFileName );
```

Comment:
I don't think this is quite true, this error message. If the file did not exists, we would not be able to map it. It should instead say something about not finding the header it expected the store file to have. I think this can still be a `StoreNotFoundException`.